xy/point_centroid.go

```diff
@@ -16,6 +16,20 @@ func PointsCentroid(point *geom.Point, extra ...*geom.Point) geom.Coord {
 	return calc.GetCentroid()
 }
 
+// MultiPointCentroid computes the centroid of the multi point argument
+//
+// Algorithm: average of all points
+func MultiPointCentroid(point *geom.MultiPoint) geom.Coord {
+	calc := NewPointCentroidCalculator()
+	coords := point.FlatCoords()
+	stride := point.Layout().Stride()
+	for i := 0; i < len(coords); i += stride {
```

Comment:
So you think the performance degradation is worth it in this case? 

I suppose you are assuming that multipoints will be relatively small sets?